File: gluecodium/src/test/resources/smoke/defaults_const/output/android/com/example/smoke/EnumDefaults.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import com.example.NativeBase;
import com.example.fire.Enum1;
import com.example.fire.Enum2;
import com.example.fire.Enum3;
import com.example.fire.Enum4;
public final class EnumDefaults extends NativeBase {
    public static final class SimpleEnum {
        @NonNull
        public Enum1 enumField;
        public SimpleEnum() {
            this.enumField = Enum1.DISABLED;
        }
    }
    public static final class NullableEnum {
        @Nullable
        public Enum2 enumField1;
        @Nullable
        public Enum2 enumField1;
        public NullableEnum() {
            this.enumField1 = null;
            this.enumField1 = Enum2.DISABLED;
        }
    }
    public static final class AliasEnum {
        @NonNull
        public Enum3 enumField;
        public AliasEnum() {
            this.enumField = Enum3.DISABLED;
        }
    }
    public static final class WrappedEnum {
        @NonNull
        public EnumWrapper structField;
        public WrappedEnum() {
            this.structField = new EnumWrapper(Enum4.DISABLED);
        }
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected EnumDefaults(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/defaults_const/output/android-kotlin/com/example/smoke/EnumDefaults.kt
/*

 *
 */

@file:JvmName("EnumDefaults")

package com.example.smoke

import com.example.NativeBase
import com.example.fire.Enum1
import com.example.fire.Enum2
import com.example.fire.Enum3
import com.example.fire.Enum4

class EnumDefaults : NativeBase {

    class SimpleEnum {
        @JvmField var enumField: Enum1



        constructor() {
            this.enumField = Enum1.DISABLED
        }





    }

    class NullableEnum {
        @JvmField var enumField1: Enum2?
        @JvmField var enumField1: Enum2?



        constructor() {
            this.enumField1 = null
            this.enumField1 = Enum2.DISABLED
        }





    }

    class AliasEnum {
        @JvmField var enumField: Enum3



        constructor() {
            this.enumField = Enum3.DISABLED
        }





    }

    class WrappedEnum {
        @JvmField var structField: EnumWrapper



        constructor() {
            this.structField = EnumWrapper(Enum4.DISABLED)
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

